UXIT-Contractors/UXIT-Planner-App
# Day/feat/notify user on absence

## Why this PR?

This PR introduces notifications for when users resign from their shift.

## What changed?

The changes in this PR affect various areas of the codebase:
   
1. **Prisma schema (`prisma/schema.prisma`):** Changes were made to the `User` and `Shift` models in the Prisma schema. The `image` field was removed from the `User` model and an `@@unique([start, end])` constraint was added to the `Shift` model.

2.  **Schedule generation (`server/api/schedule/generate-functions.ts`):** The logic for generating schedules was updated. Specifically, the condition for continuing the schedule generation loop was modified.

## How to test?

1. **Check out the branch:** Start by checking out the `feat/notify-user-on-absence` branch on your local machine.

2. **Run the application:** Seed, build and run the application.

3. **Verify notifications are enabled:** Verify notifications are enabled under the user settings.

4. **Navigate towards friday, and resign from the shift that is presented.

5. **Test the date picker functionality:** A notification should show, click it

6. **Verify a notification is sent:** A notification should appear, click the notification. This should redirect to the open-staffing website

7. **Accept the invite:** Click on accept and you should have the shift back under the correct date.
